src/inference/dev_api/openvino/runtime/threading/istreams_executor.hpp

```diff
@@ -166,12 +170,47 @@ class OPENVINO_RUNTIME_API IStreamsExecutor : virtual public ITaskExecutor {
      */
     virtual int get_numa_node_id() = 0;
 
+    /**
+     * @brief Return the id of current socket
+     * @return `ID` of current socket, or throws exceptions if called not from stream thread
+     */
+    virtual int get_socket_id() = 0;
+
     /**
      * @brief Execute the task in the current thread using streams executor configuration and constraints
      * @param task A task to start
      */
     virtual void execute(Task task) = 0;
 };
 
+enum StreamCreateType {
```

Comment:
We already have `ThreadBindingType` which has more or less the same meaning. Does it make sense to unite them?

Reply:
Yes, we planned to unite them when we clean up the MT 1.0 code. 

Now `ThreadBindingType` is part of StreamsExecutor interface of MT 1.0 and used by AUTO plugins. We will introduce StreamsExecutor interface of MT 2.0 which is based on streams_info_table and work with AUTO/GPU to upgrade.